doc/sphinx/source/preface/architecture.rst

```diff
@@ -4,7 +4,83 @@
 Architecture
 ************
 
-To be written.
+Overview and input
+==================
+
+From a functional point of view, at its core, ESMValTool needs to perform
+a number of standard diagnostics and output a collection of metrics and graphical
+objects (plots). Thus, the tool will need a certain amount of input from the user
+so that it can:
+
+* establish the correct input and output parameters and the structured workflow;
+* acquire the correct data;
+* execute the workflow and...
+* output the desired collective data and media.
+
```

Comment:
In the following paragraph, please add links to the documentation section on the config-user.yml file and on the recipe.